pkg/shared/connection/api/clustermgmt/ocmfuncs.go

```diff
@@ -0,0 +1,117 @@
+package clustermgmt
+
+import (
+	"github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+	"github.com/redhat-developer/app-services-sdk-go/kafkamgmt/apiv1/client"
+)
+
+func GetClusterList(f *factory.Factory, accessToken string, clustermgmturl string) (*v1.ClusterList, error) {
+	conn, err := f.Connection()
+	if err != nil {
+		return nil, err
+	}
+	client, cc, err := conn.API().OCMClustermgmt(clustermgmturl, accessToken)
+	if err != nil {
+		return nil, err
+	}
+	defer cc()
+	// TO-DO deal with pagination, validate clusters -- must be multi AZ and ready.
+	resource := client.Clusters().List()
+	response, err := resource.Send()
+	if err != nil {
+		return nil, err
+	}
+	clusters := response.Items()
+	return clusters, nil
+}
+
+func GetMachinePoolList(f *factory.Factory, clustermgmturl string, accessToken string, clusterId string) (*v1.MachinePoolsListResponse, error) {
+	conn, err := f.Connection()
+	if err != nil {
+		return nil, err
+	}
+	client, cc, err := conn.API().OCMClustermgmt(clustermgmturl, accessToken)
+	if err != nil {
+		return nil, err
+	}
+	defer cc()
+	resource := client.Clusters().Cluster(clusterId).MachinePools().List()
+	response, err := resource.Send()
+	if err != nil {
+		return nil, err
+	}
+	return response, nil
+}
+
+func CreateAddonWithParams(f *factory.Factory, clustermgmturl string, accessToken string, addonId string, params *[]kafkamgmtclient.FleetshardParameter, clusterId string) error {
+	conn, err := f.Connection()
```

Comment:
the following lines until `defer cc()` are repeated multiple times, can they be refactored?